src/InputParser.jl

```diff
@@ -0,0 +1,374 @@
+module InputParser
+
+# Julia implicitly numbers Enum items 0,1,2,3,4 (in order)
+# I put them in same order as FORTRAN code (i.e. NOPT=0 meant
+# ConsolidationSwell so Int(ConsolidationSwell) = 0) however
+# this wont matter since we should never be referring to the integer 
+# values explicitly anyways
+@enum Model ConsolidationSwell LeonardFrost Schmertmann CollapsibleSoil SchmertmannElastic 
+
+# These are variables/objects visible to all modules
+# that include InputParser module
+export Model, InputData
+
+# This code repeats many times in InputData constructor
+# Parses String at input[index] and returns an array A such that size(A) == items
+function parseCurrentLine(input::Array{String}, items::Int, index::Int)
+    currentLine = input[index]
+    # Tabs ruin splitting by space
+    currentLine = replace(currentLine, "\t" => " ")
+    currentLineData = split(currentLine, " ")
+    currentLineData = filter(x -> x != "", currentLineData)
+    # Check if data is there
+    if size(currentLineData)[1] < items
+        println("Error: Invalid input file!")
+        println("\t>Line $(index): $(items) values expected")
+        return -1
+    end
+    return currentLineData
+end
+
+# TODO: Search for error handling (throws Exception/ try-catch block)
+# to make InputData constructor more robust. Many Type errors are 
+# possible when converting input file
+
+# struct for InputData (OOP in Julia)
+struct InputData
+    problemName::String
+    numProblems::Int
+    model::Model
+    foundation::Bool
```

Comment:
I like the use of a bool.  What does foundation = True mean?  I'm wondering if this is a case where it is an enum with two choices, or legitimately a Boolean.

Reply:
Should I change it to an enum? It was a bit unclear since your first sentence seemed to like it being a boolean

Reply:
Based on your answer to my question, yes, you should change it to an enum.  I do like seeing a Boolean, because so many programmers seem to avoid them, especially in research software.  I liked seeing something represented as a Boolean, but then when I thought about it, I had to question whether this was actually a Boolean.  :smile:  We might add a different type of foundation in the future, and then a Boolean would be problematic.  An enum is also better for self-documenting the code.  Otherwise you have to remember whether True is rectangular or long strip.